src/qcodes/parameters/parameter_base.py

```diff
@@ -77,7 +77,17 @@ def __exit__(
             self._parameter._settable = self._original_settable
 
         if self._parameter.cache() != self._original_value:
-            self._parameter.set(self._original_value)
+            try:
+                self._parameter.set(self._original_value)
+            except Exception:
+                # Likely an uninitialized Parameter
+                vals = self._parameter._vals
+                set_parser = self._parameter.set_parser
+                self._parameter._vals = []
+                self._parameter.set_parser = None
+                self._parameter.cache.set(self._original_value)
+                self._parameter.set_parser = set_parser
+                self._parameter._vals = vals
```

Comment:
I actually think that it might be better to just leave the parameter as is if we see a warning. 

I suggest that we log the Exception here with level info 

LOG.info("some message" exc_info=True)